infra/src/main/java/com/redhat/service/bridge/infra/k8s/KubernetesResourceType.java

```diff
@@ -0,0 +1,7 @@
+package com.redhat.service.bridge.infra.k8s;
+
+public enum KubernetesResourceType {
```

Comment:
@r00ta What is this enum used for? I can see you adding it to the `ResourceEvent`, but I'm not sure what we use it for?

Reply:
@robpblake basically when a resource is updated/deleted the controllers get called
```
    void onEvent(@Observes ResourceEvent event) { // equivalent of ResourceEventSource for operator sdk
        if (event.getSubject().equals(K8SBridgeConstants.BRIDGE_TYPE)) {

            /*
             * If the CRD is deleted, remove also all the other related resource
             *
             * If another dependent resource is deleted, the `reconcileExecutor` will catch the mismatch between the expected state and the current state.
             * It will redeploy the resources so to reach the expected status at the end.
             */
            if (event.getAction().equals(Action.DELETED) && event.getResourceType().equals(KubernetesResourceType.CUSTOM_RESOURCE)) {
                delete(event.getResourceId());
                return;
            }
```
in order to properly notify that the custom resource has been deleted and that all the related resources have to be cleaned up, I had to include the `KubernetesResourceType` in the `ResourceEvent`. 

With the operator-sdk we already get called only when the CRD is deleted (see https://github.com/r00ta/baaas-decision-fleet-shard/blob/c173edb2fdd3d02abbb9e2cfc1fa23f71b99b1d6/decision-fleet-shard-operator/src/main/java/org/kie/baaas/dfs/controller/DecisionVersionController.java#L90). So our mock is not 100% matching the real api but it does not really matter since it will be thrown away and a small refactoring will be necessary.